### PR TITLE
Port AddAccessibilityModifiers analyzer/fixer/tests to shared layer

### PR DESCRIPTION
**Pull request by mavasani**
First commit - Move certain syntax querying APIs down from `SyntaxGenerator` to `ISyntaxFacts`. These APIs are used by accessibility related analyzers.
Second commit - Analyzer/Fixer/Tests file moves
Third commit - xlf file changes

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs [hunk 1]
```diff
@@ -1559,8 +1492,8 @@ private static DeclarationModifiers GetAllowedModifiers(SyntaxKind kind)
 
         public override DeclarationModifiers GetModifiers(SyntaxNode declaration)
```
Comment (mavasani) [inline src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs]: Even this API can likely move to ISyntaxFacts in future.

File: src/Analyzers/Core/CodeFixes/AddAccessibilityModifiers/AbstractAddAccessibilityModifiersCodeFixProvider.cs [hunk 2]
```diff
@@ -63,15 +69,22 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
             }
         }
 
-        private class MyCodeAction : CodeAction.DocumentChangeAction
+        private class MyCodeAction : CustomCodeActions.DocumentChangeAction
         {
+#if CODE_STYLE // 'CodeActionPriority' is not a public API, hence not supported in CodeStyle layer.
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/AddAccessibilityModifiers/AbstractAddAccessibilityModifiersCodeFixProvider.cs]: hrmm... we should add this to the list.

Comment (mavasani) [reply src/Analyzers/Core/CodeFixes/AddAccessibilityModifiers/AbstractAddAccessibilityModifiersCodeFixProvider.cs]: Filed https://github.com/dotnet/roslyn/issues/42431

**Comment by CyrusNajmabadi** [comment]
Hrmm. I don't think conditional tables keyed off of strings work. The strings stay resident and the table is never cleared

**Comment by mavasani** [comment]
> Hrmm. I don't think conditional tables keyed off of strings work. The strings stay resident and the table is never cleared

Ah, good point. I can change that to a concurrent dictionary - we only expect couple of entries per-workspace for it anyways - one for C# and one for VB. Note that this MEF based language service detection is temporary, which should hopefully be deleted soon.

**Pull request merged**